File: code/backend/php/src/Core/Domain/Story/StoryDetail.php
<?php

declare(strict_types=1);

namespace Games\Paths\Core\Domain\Story;

class StoryDetail
{
    /**
     * @param DifficultyInfo[] $difficulties
     * @param CharacterTemplateInfo[] $characterTemplates
     * @param ClassInfo[] $classes
     * @param TraitInfo[] $traits
     */
    public function __construct(
        public string $uuid,
        public ?string $title = null,
        public ?string $description = null,
        public ?string $author = null,
        public ?string $category = null,
        public ?string $group = null,
        public ?string $visibility = null,
        public int $priority = 0,
        public int $peghi = 0,
        public ?string $versionMin = null,
        public ?string $versionMax = null,
        public ?string $clockSingularDescription = null,
        public ?string $clockPluralDescription = null,
        public ?string $copyrightText = null,
        public ?string $linkCopyright = null,
        public int $locationCount = 0,
        public int $eventCount = 0,
        public int $itemCount = 0,
        public int $classCount = 0,
        public int $characterTemplateCount = 0,
        public int $traitCount = 0,
        public array $difficulties = [],
        public array $characterTemplates = [],
        public array $classes = [],
        public array $traits = [],
        public ?CardInfo $card = null
    ) {
    }
}

File: code/backend/python/app/core/models/story/story_detail.py
from dataclasses import dataclass, field
from typing import List, Optional
from app.core.models.story.difficulty_info import DifficultyInfo
from app.core.models.story.character_template_info import CharacterTemplateInfo
from app.core.models.story.class_info import ClassInfo
from app.core.models.story.trait_info import TraitInfo
from app.core.models.story.card_info import CardInfo

@dataclass
class StoryDetail:
    uuid: str
    title: str | None = None
    description: str | None = None
    author: str | None = None
    category: str | None = None
    group: str | None = None
    visibility: str | None = None
    priority: int = 0
    peghi: int = 0
    versionMin: str | None = None
    versionMax: str | None = None
    clockSingularDescription: str | None = None
    clockPluralDescription: str | None = None
    copyrightText: str | None = None
    linkCopyright: str | None = None
    locationCount: int = 0
    eventCount: int = 0
    itemCount: int = 0
    classCount: int = 0
    characterTemplateCount: int = 0
    traitCount: int = 0
    difficulties: List[DifficultyInfo] = field(default_factory=list)
    characterTemplates: List[CharacterTemplateInfo] = field(default_factory=list)
    classes: List[ClassInfo] = field(default_factory=list)
    traits: List[TraitInfo] = field(default_factory=list)
    card: Optional[CardInfo] = None
